internal/config/mount_config.go

```diff
@@ -19,6 +19,10 @@ import (
 	"time"
 )
 
+// Special type declared to stop any random string
+// from being set to MetadataPrefetchMode.
+type metadataPrefetchModeValue string
```

Comment:
How about using string as the type instead of creating a new type and add validation to return error if it's not in the specified list?

Reply:
I originally was planning to use it for string type-checking, but have now switched it to string.